### PR TITLE
Add unit test for frozen prototypes 

### PR DESCRIPTION
**Pull request by Ra1Nik**
Add unit test

### CODE REVIEW

File: tests/models/modules/test_SwaVFreezing.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    def test_swav_frozen_prototypes(self, device: str = "cpu", seed=0):
+        criterion = torch.nn.L1Loss()
+        linear_layer = torch.nn.Linear(8, 8, bias=False)
+        prototypes = SwaVPrototypes(8, 8, 2)
+        optimizer = torch.optim.SGD(prototypes.parameters(), lr=0.01)
+        for step in range(4):
+            torch.manual_seed(seed)
+            in_features = torch.rand(4, 8, device="cpu")
+            out_features = linear_layer(in_features)
+            target_features = torch.ones(4, 8, device="cpu")
```
Comment (guarin) [inline tests/models/modules/test_SwaVFreezing.py:17]: I think the features only have to be defined once. Same for setting the seed.

```suggestion
        torch.manual_seed(seed)
        in_features = torch.rand(4, 8, device="cpu")
        target_features = torch.ones(4, 8, device="cpu")
        for step in range(4):
            out_features = linear_layer(in_features)
```

File: tests/models/modules/test_SwaVFreezing.py [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+import torch
+
+from lightly.models.modules.heads import SwaVPrototypes
+
+class TestSwavFreezingPrototypes(unittest.TestCase):
+    def test_swav_frozen_prototypes(self, device: str = "cpu", seed=0):
+        criterion = torch.nn.L1Loss()
+        linear_layer = torch.nn.Linear(8, 8, bias=False)
+        prototypes = SwaVPrototypes(8, 8, 2)
```
Comment (guarin) [inline tests/models/modules/test_SwaVFreezing.py:11]: Please use argument names so it is easier to know what the `8` and `2` belong to.

File: tests/models/modules/test_SwaVFreezing.py [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+            loss = criterion(out_features, target_features)
+            loss.backward()
+            optimizer.step()
+            optimizer.zero_grad()
+            if step == 0:
+                loss0 = loss
+            if step <= 2:
+                self.assertEqual(loss, loss0)
+            if step > 2:
+                self.assertNotEqual(loss, loss0)
```
Comment (guarin) [inline tests/models/modules/test_SwaVFreezing.py:28]: Neat :)

File: tests/models/modules/test_SwaVFreezing.py [hunk 4]
```diff
@@ -0,0 +1,33 @@
+import unittest
+
+import torch
+
+from lightly.models.modules.heads import SwaVPrototypes
+
+class TestSwavFreezingPrototypes(unittest.TestCase):
+    def test_swav_frozen_prototypes(self, device: str = "cpu", seed=0):
```
Comment (guarin) [inline tests/models/modules/test_SwaVFreezing.py:8]: Could you move the test also to `test_ProjectionHeads`? Tests for functions from the same file should also always be in the same file.

File: tests/models/test_ProjectionHeads.py [hunk 5]
```diff
@@ -92,6 +92,22 @@ def test_swav_prototypes(self, device: str = 'cpu', seed=0):
                             y = prototypes(x)
                         self.assertEqual(y.shape[0], batch_size)
                         self.assertEqual(y.shape[1], n_prototypes)
+
+    def test_swav_frozen_prototypes(self, device: str = 'cpu', seed=0):
```
Comment (guarin) [inline tests/models/test_ProjectionHeads.py:96]: No need for this test, the `test_swav_prototypes` covers this already.

**Review by guarin — CHANGES_REQUESTED** [review]
Great stuff! I think you can even drop one of the tests.

**Review by guarin — APPROVED** [review]
